crates/core/src/client/mod.rs

```diff
@@ -499,12 +488,68 @@ impl<P: Provider + Send + Sync> KakarotEthApi<P> for KakarotClient<P> {
     }
 
     /// Returns the estimated gas for a transaction
-    async fn estimate_gas(
-        &self,
-        _call_request: CallRequest,
-        _block_number: Option<BlockId>,
-    ) -> Result<U256, EthApiError<P::Error>> {
-        todo!();
+    async fn estimate_gas(&self, request: CallRequest, block_id: BlockId) -> Result<U256, EthApiError<P::Error>> {
+        let chain_id: u64 = request
+            .chain_id
+            .ok_or_else(|| EthApiError::MissingParameterError("chain_id for estimate_gas".into()))?
```

Comment:
what is best between ok_or and ok_or_else
knowing that the latter takes a closure but is evaluated lazily

Reply:
I guess in this context it makes more sense to use ok_or_else since otherwise you will evaluate the conversion of the &str to a String.